### PR TITLE
Don't Reset the weather when converting a save to scenario

### PR DESCRIPTION
**Pull request by MaxMallon**
Create an option to not reset the weather when converting a save to scenario (to be able to create moody parks where it rains upon starting the scenario).

### CODE REVIEW

**Comment by IntelOrca** [comment]
It might just be better to only reset the weather when loading a .sc4/.sc6. All .park files can just keep the current weather by default.

**Comment by MaxMallon** [comment]
> It might just be better to only reset the weather when loading a .sc4/.sc6. All .park files can just keep the current weather by default.

Given that's not really the description of this PR, shall I make a new one?

**Comment by Broxzier** [comment]
You can update the title and description to match any changes made to this PR.

**Comment by MaxMallon** [comment]
> You can update the title and description to match any changes made to this PR.

Alright, I'm changing it now, but am running into a problem. The climate_reset isn't ever called for loading sc4/6 files.

It's called in scenario.cpp scenario_prepare_for_save, which is used when converting a save to scenario.
It's called in GameState.cpp, in InitAll, of which I'm not sure when it's called
It's called in ClearMapForEditing, in Editor.cpp, which I assume is when you just want to load the landscape.
And lastly, it's called in ConvertSaveToScenarioCallback, in Editor.cpp, which seems a bit double, as scenario_prepare_for_save should've already done it there anyway. 

As far as I can tell, it should definitely remain in the ClearMap one, as there you just want the terrain. It should be removed from the 2 convert-save ones, and I'm not sure about the InitAll. Is that one somewhere used in loading sc4/6?

**Comment by duncanspumpkin** [comment]
That's because you don't convert parkfiles to scenarios. They are already scenarios.

**Comment by MaxMallon** [comment]
> That's because you don't convert parkfiles to scenarios. They are already scenarios.

Ah, that would explain why sometimes unfinished scenarios show up in the scenariolist anyway. 
Either way, the changes I've made now seem to do the right thing. Right? Or did I miss something?

File: src/openrct2/park/ParkFile.cpp [hunk 1]
```diff
@@ -941,7 +941,7 @@ namespace OpenRCT2
 
                     if (cs.GetMode() == OrcaStream::Mode::READING)
                     {
-                        OpenRCT2::GetContext()->GetGameState()->InitAll(gMapSize);
+                        OpenRCT2::GetContext()->GetGameState()->InitAll(gMapSize, true);
```
Comment (IntelOrca) [inline src/openrct2/park/ParkFile.cpp:944]: Why is this necessary? `ReadWriteClimateChunk` is done after this and that will overwrite the climate anyway.

Comment (MaxMallon) [reply src/openrct2/park/ParkFile.cpp:944]: I wasn't aware of that. So then, the only thing that needs to change is the 2 removed calls of the reset function?

Comment (IntelOrca) [reply src/openrct2/park/ParkFile.cpp:944]: Yes

File: src/openrct2/scenario/Scenario.cpp [hunk 2]
```diff
@@ -608,9 +608,6 @@ bool scenario_prepare_for_save()
     if (gScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
         gParkFlags |= PARK_FLAGS_PARK_OPEN;
 
-    // Fix #2385: saved scenarios did not initialise temperatures to selected climate
-    climate_reset(gClimate);
```
Comment (Broxzier) [inline src/openrct2/scenario/Scenario.cpp:612]: This undoes [this fix](https://github.com/OpenRCT2/OpenRCT2/commit/9e203998931b6b5285d4ee43bd963f822c9dc8f5). Isn't it necessary anymore?

Comment (MaxMallon) [reply src/openrct2/scenario/Scenario.cpp:612]: I have tested it by opening several scenarios with different climates, and setting it to hyperspeed. As far as I can tell the weather changes appropriately.

File: distribution/changelog.txt [hunk 3]
```diff
@@ -43,6 +43,7 @@
 - Change: [#16710] Changed default view of Guest List to 'Thoughts' and selected tab will default to 'Summarised' (when opened from the menu).
 - Change: [#16859] Guests with umbrellas no longer always avoid going into a Maze.
 - Change: [#16912] Tired or nauseated guests will no longer jump in a Maze.
+- Change: [#17002] Weather no longer resets when converting a save to scenario.
```
Comment (Broxzier) [inline distribution/changelog.txt:46]: This needs to be moved up to the new release.

**Review by Broxzier — COMMENTED** [review]
Please rebase this PR for the next release. I'm still a bit sceptical of the removal of the "Fix #2385" part, though I believe the difference is barely if at all noticeable to the players.

**Comment by MaxMallon** [comment]
> Please rebase this PR for the next release. I'm still a bit sceptical of the removal of the "Fix #2385" part, though I believe the difference is barely if at all noticeable to the players.

What would the difference be? When I tested it I didn't really notice any difference, even when specifically looking for climate-related differences

**Review by Broxzier — APPROVED** [review]

**Review by tupaschoal — COMMENTED** [review]
Can you please rebase to fix the merge conflicts?
